### PR TITLE
Spin multi version

### PR DESCRIPTION
**Pull request by karthik2804**

### CODE REVIEW

File: templates/spin_sidebar.hbs [hunk 1]
```diff
@@ -1,229 +1,293 @@
...
-                    <li><a {{#if (active_content request.spin-path-info "/spin/url-shortener-tutorial" )}} class="active" {{/if}}
-                            href="{{site.info.base_url}}/spin/url-shortener-tutorial">Building a URL Shortener With Spin</a></li>
-                    <li><a {{#if (active_content request.spin-path-info "/spin/registry-tutorial" )}} class="active" {{/if}}
-                            href="{{site.info.base_url}}/spin/registry-tutorial">Spin Apps in Registries</a></li>
-                    <li><a {{#if (active_content request.spin-path-info "/spin/key-value-store-tutorial" )}} class="active"
-                            {{/if}} href="{{site.info.base_url}}/spin/key-value-store-tutorial">Spin Key-Value Store</a></li>
+                        <li><a {{#if (active_project request.spin-full-url "/spin/v1/serverless-ai-hello-world" )}}
+                                        class="active" {{/if}}
+                                        href="{{site.info.base_url}}/spin/serverless-ai-hello-world.md">Build Your First
```
Comment (tpmccallum) [inline templates/spin_sidebar.hbs:147]:
```suggestion
                                        href="{{site.info.base_url}}/spin/v1/serverless-ai-hello-world">Build Your First
```

File: templates/spin_sidebar_v2.hbs [hunk 2]
```diff
@@ -0,0 +1,292 @@
+<div class="accordion-tabs">
+        <div class="accordion-menu-item">
+                <input type="checkbox" id="rd0" name="rd">
+                <label for="rd0" class="accordion-menu-item-label menu-label">
+                        Get Started
+                </label>
+                <ul class="menu-list accordion-menu-item-content">
+                        <li><a {{#if (active_project request.spin-full-url "/spin/v2/serverless-ai-hello-world" )}}
+                                        class="active" {{/if}}
+                                        href="{{site.info.base_url}}/spin/serverless-ai-hello-world.md">Build Your First
```
Comment (tpmccallum) [inline templates/spin_sidebar_v2.hbs:10]:
```suggestion
                                        href="{{site.info.base_url}}/spin/v2/serverless-ai-hello-world.md">Build Your First
```

File: templates/content_navbar.hbs [hunk 3]
```diff
@@ -62,17 +62,17 @@
                 {{/if}}
                 {{/if}}
                 {{/if}}
-                
+
                 <div class="navbar-dropdown is-boxed">
-                    <a href="/spin/index" class="navbar-item{{#if (active_project request.spin-path-info "/spin/" )}}
+                    <a href="/spin/index" class="navbar-item{{#if (active_project request.spin-path-info " /spin/" )}}
```
Comment (tpmccallum) [inline templates/content_navbar.hbs:67]:
```suggestion
                    <a href="/spin/index" class="navbar-item{{#if (active_project request.spin-path-info "/spin/" )}}
```

File: spin-redirecter/src/index.ts [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+        location: routeMatch
+      }
+    }
+  }
+
+  // If none of the specified redirects matches just convert to latest version
+  return {
+    status: 302,
+    headers: {
+      location: reqPath.replace("/spin", `/spin/${latest_spin_version}`)
```
Comment (tpmccallum) [inline spin-redirecter/src/index.ts:32]: Should this be `.replace("/spin/",` instead of just `/spin` (do we need to add a trailing slash)?

**Review by tpmccallum — COMMENTED** [review]
Just a first pass at it.
Will take another pass now. And incorporate local preview and testing.

File: content/spin/v1/serverless-ai-hello-world.md [hunk 5]
```diff
@@ -0,0 +1,390 @@
...
+
+* Install Spin (and dependencies) on your local machine
+* Create a ‘Hello World’ Serverless AI application
+* Learn about the Serverless AI SDK (in Rust, TypeScript and Python)
+
+## Tutorial Prerequisites
+
+### Spin
+
+<<<<<<< HEAD:content/_spin/v1/serverless-ai-hello-world.md
```
Comment (tpmccallum) [inline content/spin/v1/serverless-ai-hello-world.md:34]: Please remove this inline conflict @karthik2804 ?

**Review by tpmccallum — CHANGES_REQUESTED** [review]

File: spin-redirecter/src/index.ts (hunk 4 above)
Comment (karthik2804) [reply spin-redirecter/src/index.ts:32]: good catch

**Review by tpmccallum — APPROVED** [review]

**Review by tpmccallum — CHANGES_REQUESTED** [review]
Getting an error after re-checking it out (refreshing the latest changes) and running again on localhost.

**Comment by tpmccallum** [comment]
Screenshot 2023-10-27 at 14 19 07

Something has happened in the last few commits because after a re-pull it broke.

**Review by tpmccallum — APPROVED** [review]
LGTM
Thank you @karthik2804 👍

**Pull request merged**